tests/integ/scala/test_dataframe_reader_suite.py

```diff
@@ -395,6 +395,23 @@ def test_read_csv_with_infer_schema(session, mode, parse_header):
     Utils.check_answer(df, [Row(1, "one", 1.2), Row(2, "two", 2.2)])
 
 
+@pytest.mark.skipif(
+    "config.getoption('local_testing_mode', default=False)",
+    reason="SNOW-1435112: csv infer schema option is not supported",
+)
+@pytest.mark.parametrize("mode", ["select", "copy"])
+def test_read_csv_with_infer_schema_options(session, mode):
+    reader = get_reader(session, mode)
+    df = (
+        reader.option("INFER_SCHEMA", True)
+        .option(
+            "INFER_SCHEMA_OPTIONS", {"MAX_FILE_COUNT": 1, "MAX_RECORDS_PER_FILE": 10}
```

Comment:
does it have any effect in the test? If not, can we test it?

Reply:
Good call, changed to `IGNORE_CASE` which we can verify the effect of